internal/provider/role_data_source_test.go

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func TestAccRoleDataSource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Read testing
+			{
+				Config: providerConfig + `data "sonatypeiq_role" "role_by_name" {
+					name = "Developer"
+				}`,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.sonatypeiq_role.role_by_name", "name", "Developer"),
+					resource.TestCheckResourceAttr("data.sonatypeiq_role.role_by_name", "id", "1da70fae1fd54d6cb7999871ebdb9a36"),
```

Comment:
I hope this is a stable ID. If not, we'll have to create a dedicated role for TF provider testing. I would have loved to also create a role resource but the API does not seem to support role creation, so it'll have to be manual.

Reply:
Yeah - that won't be stable @iverberk - for now, best to remove line 36 from the check.

Reply:
Well - I wouldn't expect it to be stable, but I do appear to have the same ID in my install?

Let's try it as is 👍

Reply:
You beat me to it. I actually pushed a commit to remove it, but after your merge :-)